[BUG] Array function array_distinct returns elements in wrong order

**Failing test:** `tests/parity/functions/test_array.py::TestArrayFunctionsParity::test_array_distinct`

**Issue:**
`array_distinct` function returns array elements in a different order than PySpark, causing parity test failure.

**Error:**
```
AssertionError: DataFrames are not equivalent:
Value mismatch in column 'array_distinct(tags)' row 0[0]: mock='data', expected='backend'
Value mismatch in column 'array_distinct(tags)' row 1[0]: mock='backend', expected='data'
```

**Expected behavior:**
- `array_distinct` should return distinct elements in the same order as PySpark
- Row 0 first element should be 'backend'
- Row 1 first element should be 'data'

**Actual behavior:**
- Elements are in reverse order compared to PySpark
- Row 0 first element is 'data' (should be 'backend')
- Row 1 first element is 'backend' (should be 'data')

**Repro:**
```bash
pytest tests/parity/functions/test_array.py::TestArrayFunctionsParity::test_array_distinct -q
```

**Note:**
- This is an array function parity issue, not SQL-related
- Function works correctly (removes duplicates), but element ordering differs
- May be related to how Polars handles array distinct vs PySpark

--- a/sparkless/backend/polars/expression_translator.py
+++ b/sparkless/backend/polars/expression_translator.py
@@ -2112,7 +2112,10 @@ class PolarsExpressionTranslator:
             "size": lambda e: e.list.len(),
             "array_max": lambda e: e.list.max(),
             "array_min": lambda e: e.list.min(),
-            "array_distinct": lambda e: e.list.unique(),
+            "array_distinct": lambda e: e.map_elements(
+                lambda arr: list(dict.fromkeys(arr)) if isinstance(arr, list) else arr,
+                return_dtype=pl.List(pl.Utf8),
+            ),
             # Note: explode/explode_outer expressions just return the array column
             # The actual row expansion is handled in operation_executor
             "explode": lambda e: e,  # Return the array column as-is, will be exploded in operation_executor

